Funcky.Analyzer/Funcky.Analyzer/SyntaxMatcher.cs

```diff
@@ -0,0 +1,48 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Funcky.Analyzer
+{
+    public class SyntaxMatcher
+    {
+        private const string DefaultValue = "NIL";
+        private readonly SyntaxNodeAnalysisContext _analysisContext;
+
+        public SyntaxMatcher(SyntaxNodeAnalysisContext analysisContext)
+            => _analysisContext = analysisContext;
+
+        private InvocationExpressionSyntax InvocationExpr => (InvocationExpressionSyntax)_analysisContext.Node;
+
+        private SemanticModel SemanticModel => _analysisContext.SemanticModel;
+
+        public bool MatchStaticCall(string className, string methodName)
+            => InvocationExpr.Expression is MemberAccessExpressionSyntax memberAccessExpr
+                && memberAccessExpr.Name.Identifier.ValueText == methodName
```

Comment:
Hmm comparing just by name here could cause false positives. We only want to warn if the method on `System.Linq.Enumerable` is called, not any class called `Enumerable` imo.

You can use operations to check for actual types. See: [`GetOperation`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.semanticmodel.getoperation?view=roslyn-dotnet-3.10.0), [`IInvocationOperation `](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.operations.iinvocationoperation?view=roslyn-dotnet-3.10.0)

Reply:
Use the semantic model to check for the correct type: c881e52bccabb0f1987549a399d503d7f85c0486